ci_tutorial/products_pipeline/full_pipeline.rst

```diff
@@ -0,0 +1,303 @@
+
+.. code-block:: bash
+    :caption: Promoting from products->develop
+
+    # Promotion using Conan download/upload commands 
+    # (slow, can be improved with art:promote custom command)
+    $ conan download --list=uploaded.json -r=products --format=json > promote.json
+    $ conan upload --list=promote.json -r=develop -c
+
+
+And our final ``develop`` repository state will be:
+
+
+.. graphviz::
+    :align: center
+
+    digraph repositories {
+        node [fillcolor="lightskyblue", style=filled, shape=box]
+        rankdir="LR"; 
+        subgraph cluster_0 {
+          label="Packages server";
+          style=filled;
+          color=lightgrey;
+          subgraph cluster_1 {
+            label = "packages\n repository" 
+            shape = "box";
+            style=filled;
+            color=lightblue;
+            "packages" [style=invis];
+            "ai/1.1.0\n (Release)";
+            "ai/1.1.0\n (Debug)";
+          }
+          subgraph cluster_2 {
+            label = "products\n repository" 
+            shape = "box";
+            style=filled;
+            color=lightblue;
+            "products" [style=invis];
+          } 
+          subgraph cluster_3 {
+            rankdir="BT";
+            shape = "box";
+            label = "develop repository";
+            color=lightblue;
+            rankdir="BT";
+    
+            node [fillcolor="lightskyblue", style=filled, shape=box]
+            "game/1.0" -> "engine/1.0" -> "ai/1.0" -> "mathlib/1.0";
+            "engine/1.0" -> "graphics/1.0" -> "mathlib/1.0";
+            "mapviewer/1.0" -> "graphics/1.0";
+            "game/1.0" [fillcolor="lightgreen"];
+            "mapviewer/1.0" [fillcolor="lightgreen"];
+            "ai/promoted" [label="ai/1.1.0\n(new version)"];
+            "engine/promoted" [label="engine/1.0\n(new binary)"];
+            "game/promoted" [label="game/1.0\n(new binary)", fillcolor="lightgreen"];
+            "game/promoted" -> "engine/promoted" -> "ai/promoted" -> "mathlib/1.0";
+             "engine/promoted" -> "graphics/1.0";
+          }
+          {
+            edge[style=invis];
+            "packages" -> "products" -> "game/1.0" ; 
+            rankdir="BT";    
+          }
+        }
+    }
+
+
+This state of the ``develop`` repository will have the following behavior:
+
+- Developers installing ``game/1.0`` or ``engine/1.0`` will by default resolve to latest ``ai/1.1.0`` and use it. They will find pre-compiled binaries for the dependencies too, and they can continue developing using the latest set of dependencies.
+- Developers and CI that were using a lockfile that was locking ``ai/1.0`` version, will still be able to keep working with that dependency without anything breaking, as the new versions and package binaries do not break or invalidate the previous existing binaries.
```

Comment:
> I think the doc may also clarify what happens with the lockfile later in this workflow...

That is a good question. I will add some lines to discuss this, but in short, I don't think there is a unique good approach for this, and it might depend on many other business and development factors. We will also try to elaborate a bit more on future sections of the docs.

> Is it supposed to be updated later, once the Important People decide that updating game to use ai/1.1.0 is a Good Idea?

The ``conan.lock`` used at this point has already been updated. It contains the ``ai/1.1.0``. So no scripting magic necessary at this point.

I have added a couple of paragraphs commenting this, but still very general, this will most likely get further details in the future.